### PR TITLE
Adds two sig-cluster-lifecycle projects

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuckh@vmware.com>

Adding two new subprojects. I will need the repos created before I can link to the owner files, unless I'm going about this the wrong way?

cc @detiber 

edit: after this I can make an issue to make a repo that references this change

### CODE REVIEW

File: sigs.yaml [hunk 1]
```diff
@@ -937,6 +937,8 @@ sigs:
       slack: bootkube
     owners:
     - https://raw.githubusercontent.com/kubernetes-incubator/bootkube/master/OWNERS
+  - name: bootstrap-provider-kubeadm
```
Comment (detiber) [inline sigs.yaml]: should this subproject be prefixed with `cluster-api` to avoid confusion?

Comment (chuckha) [reply sigs.yaml]: explicit is better than implicit! will update.

File: sigs.yaml [hunk 2]
```diff
@@ -947,12 +947,16 @@ sigs:
       slack: cluster-api
     owners:
     - https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/master/OWNERS
+  - name: cluster-api-bootstrap-provider-kubeadm
+    owners: []
```
Comment (cblecker) [inline sigs.yaml]:
```suggestion
    owners:
    - https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/master/OWNERS
```

File: sigs.yaml [hunk 3]
```diff
@@ -947,12 +947,16 @@ sigs:
...
   - name: cluster-api-provider-aws
     owners:
     - https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-aws/master/OWNERS
   - name: cluster-api-provider-digitalocean
     owners:
     - https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-digitalocean/master/OWNERS
+  - name: cluster-api-provider-docker
+    owners: []
```
Comment (cblecker) [inline sigs.yaml]:
```suggestion
    owners:
    - https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-docker/master/OWNERS
```

File: sigs.yaml (hunk 2 above)
Comment (chuckha) [reply sigs.yaml]: oh, is it ok that that repo doesn't exist yet? I assumed this would be a back-and-forth dance 😅

**Comment by chuckha** [comment]
@neolit123 can you explain the hold please? I don't mind that you added a hold but it would be nice to know why.

**Comment by nikhita** [comment]
I can't speak for why the hold was added, but we usually merge the sigs.yaml PR after the repo has been created, just to be sure that the repos mentioned here really exist :grimacing:

**Comment by chuckha** [comment]
oh! thanks @nikhita I'll go open an issue to get the repos created, I thought this had to come first

**Comment by nikhita** [comment]
> I thought this had to come first

There isn't a strict order or anything. It's actually nice to have this PR out and have the required lgtms already though. :smile:

**Comment by neolit123** [comment]
Feel free to unhold if the creation was discussed on sig level.

**Comment by chuckha** [comment]
@neolit123 thank you for the clarification

**Comment by chuckha** [comment]
Evidently this is the wrong place to make a PR. What I want is a new project owned by the Cluster API group which is part of sig cluster lifecycle -- I'll close this and open a PR in kubernetes/org and update Cluster API's owners files and use those.

**Comment by nikhita** [comment]
/reopen

@chuckha can you update this to just the cluster-api-provider-docker repo?
Ref: https://github.com/kubernetes/org/issues/938

**Comment by nikhita** [comment]
Oh lol, let's move to another PR then.

**Pull request closed without merging**